### PR TITLE
Check ImgInfo data from cache. Update ImgInfo method names.

### PR DESCRIPTION
**Pull request by bcail**

### CODE REVIEW

**Comment by bcail** [comment]
@alexwlchan could you please take a look at this when you get a chance?

File: loris/img_info.py [hunk 1]
```diff
@@ -315,8 +314,8 @@ def sizes_for_scales(self, scales):
     def scale_dim(dim_len, scale):
         return int(ceil(dim_len * 1.0/scale))
 
-    def to_dict(self):
-        logger.debug('self.ident in to_dict: %s', self.ident)
+    def _get_iiif_info(self):
+        #returns only IIIF info (not Loris-specific info like src_format)
```
Comment (alexwlchan) [inline loris/img_info.py]: This should be a docstring, not a comment.

File: loris/img_info.py [hunk 2]
```diff
@@ -337,21 +337,18 @@ def to_dict(self):
...
-        if cache:
-            # Add in internal properties for caching
-            d['_src_img_fp'] = self.src_img_fp
-            d['_src_format'] = self.src_format
-            d['_auth_rules'] = self.auth_rules
+    def to_iiif_json(self):
+        d = self._get_iiif_info()
+        return json.dumps(d)
 
+    def to_full_info_json(self):
```
Comment (alexwlchan) [inline loris/img_info.py:344]: Might be worth a brief docstring explaining this method is creating the data that gets cached in the InfoCache.

File: loris/webapp.py [hunk 3]
```diff
@@ -517,7 +517,10 @@ def _get_info(self,ident,request,base_uri):
         else:
             in_cache = False
 
-        if in_cache:
+        #Checking for src_format in ImageInfo signals that it's not old cache data:
+        #   src_format should be in the cache now.
+        #   TODO: remove in Loris 4.0.
+        if in_cache and self.info_cache[request][0].src_format:
```
Comment (alexwlchan) [inline loris/webapp.py]: A few questions:

* The old cache data was `info.to_json(cache=True)`, which would presumably include `src_format`? So is this addressing an earlier time where it wasn't included?

* As Loris is 2.1.0 now, next major release is 3.0 – so why wait until 4.0 to make breaking cache changes?

* To be clear: what we'd be removing in 4.0 is the second half of this check? Maybe that should be explicit in the comment?

Meta-comment: I didn't immediately recognise the format of `self.info_cache[request]` – I wonder whether we could store namedtuples in InfoCache for greater clarity here?

**Review by alexwlchan — COMMENTED** [review]
Disambiguating `to_iiif_json()` and `to_full_info_json()` is a nice change. A few questions inline, but nothing too major.

File: loris/webapp.py (hunk 3 above)
Comment (bcail) [reply loris/webapp.py]: @alexwlchan thanks for your review. I pushed another commit. As far as whether to remove the cache check in 3.0 or 4.0, @giorgiosironi had a comment about it in https://github.com/loris-imageserver/loris/issues/356#issuecomment-330505838. @giorgiosironi what are your thoughts about removing the cache check in a 3.0 or a 4.0 release?

I'm OK either way. Regardless, we'll have to let users know that they'll have to make sure their cache doesn't have any of the old content once we remove the check.

Comment (alexwlchan) [reply loris/webapp.py]: Ah, I’d missed the idea of making 3.0 a Python 3-compatible release, and deferring the cache changes until later. I don’t mind too much when we change it, I just wanted to check if I’d forgotten another conversation, and I had.

Important thing is that we have a note to come back and change this later, which we do! :-)

**Review by alexwlchan — APPROVED** [review]
Okay, LGTM. And… ✨ merging ✨ .

**Pull request merged**